doc/source/cluster/guide.rst

```diff
@@ -0,0 +1,275 @@
+  may cause significant performance problems.
+* Use NVMe SSDs (or other high perforfmance storage) if possible. If :ref:`object
+  spilling <object-spilling>` is enabled Ray will spill objects to disk if necessary. This is
+  most commonly needed for data processing workloads.
+
+Configuring the head node
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In addition to the above changes, when deploying a large cluster, Ray's
+architecture means that the head node will have extra stress due to GCS.
+
+* Make sure the head node has sufficient bandwidth. The most heavily stressed
+  resource on the head node is outbound bandwidth. For large clusters (see the
+  scalability envelope), we recommend using machines networking characteristics
+  at least as good as an r5dn.16xlarge on AWS EC2.
+* Set ``resources: {"CPU": 0}`` on the head node. Due to the heavy networking
+  load (and the GCS and redis processes), we recommend setting the number of
+  CPUs to 0 ohn the head node to avoid scheduling additional tasks on it.
+
+Picking nodes
+^^^^^^^^^^^^^
+
+Here are some tips for how to set your ``available_node_types`` for a cluster,
+using AWS instance types as a concrete example.
+
+General recommendations with AWS instance types:
+
+**When to use GPUs**
+
+* If you’re using some RL/ML framework
+* You’re doing something with tensorflow/pytorch/jax (some framework that can
+  leverage GPUs well)
+
+**What type of GPU?**
+
+* The latest gen GPU is almost always the best bang for your buck (p3 > p2, g4
+  > g3), for most well designed applications the performance outweighs the
+  price (the instance price may be higher, but you’ll use the instance for less
+  time.
+* You may want to consider using older instances if you’re doing dev work and
+  won’t actually fully utilize the GPUs though.
+* If you’re doing training (ML or RL), you should use a P instance. If you’re
+  doing inference, you should use a G instance. The difference is
+  processing:VRAM ratio (training requires more memory).
+
+**What type of CPU?**
+
+* Again stick to the latest generation, they’re typically cheaper and faster.
+* When in doubt use M instances, they have typically have the highest
+  availability.
+* If you know your application is memory intensive (memory utilization is full,
+  but cpu is not), go with an R instance
+* If you know your application is CPU intensive go with a C instance
+* If you have a big cluster, make the head node an instance with an n (r5dn or
+  c5n)
+
+**How many CPUs/GPUs?**
+
+* Focus on your CPU:GPU ratio first and look at the utilization (Ray dashboard
+  should help with this). If your CPU utilization is low add GPUs, or vice
+  versa.
+* The exact ratio will be very dependent on your workload.
+* Once you find a good ratio, you should be able to scale up and and keep the
+  same ratio.
+* You can’t infinitely scale forever. Eventually, as you add more machines your
+  performance improvements will become sub-linear/not worth it. I don’t think
```

Comment:
Avoid "I" here. Just say it in 3rd person -- "There may not be a one-size-fits-all strategy."